src/main/java/io/aiven/kafka/connect/http/sender/AccessTokenHttpSender.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright 2021 Aiven Oy and http-connector-for-apache-kafka project contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.aiven.kafka.connect.http.sender;
+
+import java.net.URLEncoder;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.nio.charset.StandardCharsets;
+import java.time.Duration;
+import java.util.Base64;
+import java.util.StringJoiner;
+
+import org.apache.kafka.connect.errors.ConnectException;
+
+import io.aiven.kafka.connect.http.config.HttpSinkConfig;
+
+import static io.aiven.kafka.connect.http.config.OAuth2AuthorizationMode.HEADER;
+import static io.aiven.kafka.connect.http.config.OAuth2AuthorizationMode.URL;
+
+class AccessTokenHttpSender extends AbstractHttpSender implements HttpSender {
+
+    AccessTokenHttpSender(final HttpSinkConfig config, final HttpClient httpClient) {
+        super(config, new AccessTokenHttpRequestBuilder(), httpClient);
+    }
+
+    HttpResponse<String> call() {
```

Comment:
Why do we need the call method if send already exists? We could move the body builder to the existing `AccessTokenHttpRequestBuilder`, wdyt?

Reply:
Actually I don't have other alternatives in mind. The fact is the `AccessTokenHttpSender` is handling its own message by building the oauth2 form and I think we don't want the `Oauth2HttpSender` to do anything about that.
More details with an example:
If I only use the `send(message)` method I Would have to build the message (oauth2 form) in the caller :
```
private String getAccessToken() {
            // Re-use the access token if it's already defined
            if (this.accessToken != null) {
                return this.accessToken;
            }
            LOGGER.info("Configure OAuth2 for URI: {} and Client ID: {}", config.oauth2AccessTokenUri(),
                config.oauth2ClientId());
            try {
            final var accessTokenRequestBodyBuilder = new StringJoiner("&");
                    accessTokenRequestBodyBuilder.add(encodeNameAndValue("grant_type", "client_credentials"));
                    if (config.oauth2ClientScope() != null) {
                        accessTokenRequestBodyBuilder.add(encodeNameAndValue("scope", config.oauth2ClientScope()));
                    }
                    setClientIdAndSecret(accessTokenRequestBodyBuilder, config);               

                final var response = accessTokenHttpSender.send(accessTokenRequestBodyBuilder.toString());
                accessToken = buildAccessTokenAuthHeader(response.body());
            } catch (final IOException e) {
                throw new ConnectException("Couldn't get OAuth2 access token", e);
            }
            return accessToken;
        }
```

But I'm trying to segregate responsibilities so I don't see a cleaner way to do this. It does make more sense in the second PR -> https://github.com/aiven/http-connector-for-apache-kafka/pull/172/commits/968d32fd9b051f93e200e12b38250f00863e747c#diff-0eb7ade1c54d1f1714e153a7fb9a8f8b9a900b248210eddb464c41b72699eee4
I'm opened to other ideas :)